EIPS/eip-3290.md

```diff
@@ -0,0 +1,172 @@
+* Custom fallback consensus and validation logic
+* Multi TBS setup
+* NFT transfers 
+
+We also considered a solution that triggers the tokens transfer. Still, after long consideration, we decided that it should be implemented on the application layer because we wanted to leave a more extensive range of integration with this standard. 
+
+## Backwards Compatibility
+This standard is compatible with any fungible and non-fungible token, such as ERC-20 or ERC-721.
+
+## Reference Implementation
+https://github.com/Omodaka9375/Time-Based-Switch-on-Ethereum/blob/main/contracts/TimeBasedSwitch.sol
+
+## Security Considerations
+
+### Logic Bugs
+Mitigation against logic bugs:
+* Following latest Solidity coding standards
+* Following general best practices for software development
+* Using OpenZeppelin standard library (Ownership, Lifecycle, ReentrancyGuard)
+* Keeping contract logic very simple, and abstracting complicated methods to the front-end
+
+### Recursive Calls
+* Adopting modify first, pay last design pattern
+* Checking that any single input can only result in a single payment
+
+### Integer Arithmetic Overflow
+* Validating that adding a new switch to existing account wouldn't cause an overflow
+* Validating that adding a new beneficiary/executor wouldn't cause an overflow
+* Validating that adding new tokens wouldn't cause an overflow
+* No user inputted variables affect balance tracking
+* All arithmetic operations reviewed (4+'s, 5-'s)
+
+### Poison Data
+Avoiding malicious user input:
+* Adding `onlyAllowed` checks on privileged queries to the switch
+* Adding `onlyExecutorsOrOwner` privileged actions to the switch
+* Validate `switch` whenever the user inputs it
+* Validate `amount` whenever the user inputs it
+* Validate that a specified switch `time` parameter is set to minimum 1 day or 86400 seconds
+
+### Exposed Functions
+There is no exposed withdrawal function that can "drain" the switch.
+Once created funds can't be taken out of the switch, only added on top.
+Switch can be terminated early by the owner, in which case the funds are returned back to the owner.  
+
+All contract functions which are public have appropriate checks for:
+* Contract Owner
+* Switch Owner
+* Switch Executor
+* Switch beneficiary
+
+### Exposed Secrets
+TBS does not store any secret information on the blockchain.
+
+### Miner Vulnerabilities
+TBS uses 'block.timestamp' to track and schedule transactions in the future only. The switch lifetime is defined as a minimum of 1 day. There are no current security considerations regarding outside dependencies for this core feature.
+
+### Gas Limit
+Avoiding gas limit issues:
+* Not using loops in TBS
+* No strings, only byte32
+* No SafeMath to reduce contract size and dependencies
+* All searches are in constant time or O(1), with the exception of ERC20 tokens, where performance is in proportion to the number of tokens added to the switch or O(N).
```

Comment:
This section should describe gotchas and known risks that implementers need to watch out for.  Currently, this looks more like the result of a business risk analysis or an implementation audit report, which isn't appropriate for an EIP.